controllers/perconaserverformysql_controller.go

```diff
@@ -165,6 +172,193 @@ func (r *PerconaServerForMySQLReconciler) ensureUserSecrets(
 	return nil
 }
 
+func (r *PerconaServerForMySQLReconciler) reconcileUsers(ctx context.Context, cr *apiv2.PerconaServerForMySQL) error {
+	l := log.FromContext(ctx).WithName("reconcileUsers")
+
+	if cr.Status.MySQL.State != apiv2.StateReady {
+		l.Info("MySQL is not ready")
+		return nil
+	}
+
+	secret := &corev1.Secret{}
+	nn := types.NamespacedName{Name: cr.Spec.SecretsName, Namespace: cr.Namespace}
+	if err := r.Client.Get(ctx, nn, secret); err != nil {
+		return errors.Wrapf(err, "get Secret/%s", nn.Name)
+	}
+
+	internalSecret := &corev1.Secret{}
+	nn.Name = cr.InternalSecretName()
+	err := r.Client.Get(ctx, nn, internalSecret)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "get Secret/%s", nn.Name)
+	}
+
+	// Internal secret is not found
+	if k8serrors.IsNotFound(err) {
+		secret.DeepCopyInto(internalSecret)
+		internalSecret.ObjectMeta = metav1.ObjectMeta{
+			Name:      cr.InternalSecretName(),
+			Namespace: cr.Namespace,
+		}
+
+		if err := r.Client.Create(ctx, internalSecret); err != nil {
+			return errors.Wrapf(err, "create Secret/%s", internalSecret.Name)
+		}
+
+		return nil
+	}
+
+	operatorPass, err := k8s.UserPassword(ctx, r.Client, cr, apiv2.UserOperator)
+	if err != nil {
+		return errors.Wrap(err, "get operator password")
+	}
+
+	orcHost := orchestrator.APIHost(orchestrator.ServiceName(cr))
+	primary, err := orchestrator.ClusterPrimary(ctx, orcHost, cr.ClusterHint())
+	if err != nil {
+		return errors.Wrap(err, "get cluster primary")
+	}
+	l.V(1).Info("Got cluster primary", "primary", primary)
+	primaryHost := getPrimaryHostname(primary, cr)
+
+	um, err := users.NewManager(apiv2.UserOperator, operatorPass, primaryHost, mysql.DefaultAdminPort)
```

Comment:
case: it is needed to change multiple passwords in a loop
what if primary will be changed during monitor password update?

Reply:
You're right, I was doing multiple API requests to k8s and orchestrator between each password update. In https://github.com/percona/percona-server-mysql-operator/pull/26/commits/4e9700b50fab2c7a80c49ea9ad6c19515f7d984b I changed the implementation to update all users in a single transaction as in PXC operator.